### PR TITLE
DHFPROD-5793: Find entity type references in referenced steps

### PR DESCRIPTION
**Pull request by akshaysonvane**
### Description
* Assigned `data-hub-match-merge-reader` and `data-hub-custom-reader` roles to `hub-central-entity-model-writer` role so that we can check for entity type references in matching/merging/mastering/custom steps.
* Changed JSON model key from `stepAndMappingNames` to `stepNames`.

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/entity-lib.sjs [hunk 1]
```diff
@@ -218,31 +218,12 @@ function deleteModel(entityName) {
  * @returns {[]}
  */
 function findModelReferencesInSteps(entityName, entityTypeId) {
-  let steps = [];
-
-  const mappingQuery = cts.andQuery([
-    cts.collectionQuery('http://marklogic.com/data-hub/mappings'),
-    cts.jsonPropertyValueQuery("targetEntityType", [entityName, entityTypeId])
+  const stepQuery = cts.andQuery([
+    cts.collectionQuery('http://marklogic.com/data-hub/steps'),
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/entity-lib.sjs:222]: I think this fixes another bug which is that the original query was querying on the mappings collection, but that will pick up existing legacy mappings that have not yet been deleted. And we don't care about those.

File: marklogic-data-hub-central/ui/e2e/hc-qa-project/build.gradle [hunk 2]
```diff
@@ -1,6 +1,5 @@
 buildscript {
     repositories {
-        maven {url 'http://distro.marklogic.com/nexus/repository/maven-snapshots/'}
```
[the remark this answers is not in the document]
Comment (bsrikan) [reply marklogic-data-hub-central/ui/e2e/hc-qa-project/build.gradle:3]: @SameeraPriyathamTadikonda how about the tests that run on DHS. Do we publish to maven local for those tests as well.

**Comment by SameeraPriyathamTadikonda** [comment]
Are the tests reviewed? Can I check the box so that PR gets merged?

File: marklogic-data-hub-central/ui/e2e/hc-qa-project/build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (SameeraPriyathamTadikonda) [reply marklogic-data-hub-central/ui/e2e/hc-qa-project/build.gradle:3]: I've removed this commit. We will be using nexus for tests that run on DHS.

[the remark this answers is not in the document]
Comment (bsrikan) [reply marklogic-data-hub-central/ui/e2e/hc-qa-project/build.gradle:3]: This code is now merged with nexus repo removed from build.gradle

**Pull request merged**